components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/JDBCSubscriberDAOImpl.java

```diff
@@ -0,0 +1,372 @@
+            populateProperties(holder, oldHolder);
+        } else {
+            throw new EntitlementException("Subscriber ID does not exist; update cannot be done");
+        }
+
+        PublisherPropertyDTO[] propertyDTOs = holder.getPropertyDTOs();
+        updateSubscriber(subscriberId, holder, propertyDTOs, oldHolder);
+    }
+
+    /**
+     * Removes the subscriber of the given subscriber ID.
+     *
+     * @param subscriberId subscriber ID.
+     * @throws EntitlementException If an error occurs.
+     */
+    @Override
+    public void removeSubscriber(String subscriberId) throws EntitlementException {
+
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+
+        if (StringUtils.isBlank(subscriberId)) {
+            throw new EntitlementException(ERROR_SUBSCRIBER_ID_NULL);
+        }
+
+        if (EntitlementConstants.PDP_SUBSCRIBER_ID.equals(subscriberId.trim())) {
+            throw new EntitlementException("Cannot delete PDP publisher");
+        }
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (NamedPreparedStatement preparedStmt = new NamedPreparedStatement(connection, DELETE_SUBSCRIBER_SQL)) {
+                preparedStmt.setString(SUBSCRIBER_ID, subscriberId);
+                preparedStmt.setInt(TENANT_ID, tenantId);
+                preparedStmt.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new EntitlementException("Error while deleting subscriber details", e);
+        }
+    }
+
+    /**
+     * Checks whether a subscriber exists.
+     *
+     * @param subscriberId subscriber ID.
+     * @return whether the subscriber exists or not.
+     * @throws EntitlementException If an error occurs.
+     */
+    public boolean isSubscriberExists(String subscriberId) throws EntitlementException {
+
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (NamedPreparedStatement findSubscriberExistencePrepStmt = new NamedPreparedStatement(connection,
+                    GET_SUBSCRIBER_EXISTENCE_SQL)) {
+                findSubscriberExistencePrepStmt.setString(SUBSCRIBER_ID, subscriberId);
+                findSubscriberExistencePrepStmt.setInt(TENANT_ID, tenantId);
+
+                try (ResultSet rs1 = findSubscriberExistencePrepStmt.executeQuery()) {
+                    return rs1.next();
+                }
+            }
+        } catch (SQLException e) {
+            throw new EntitlementException("Error while checking subscriber existence", e);
+        }
+    }
+
+    private void insertSubscriber(String subscriberId, PublisherDataHolder holder, PublisherPropertyDTO[] propertyDTOs)
+            throws EntitlementException {
+
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
```

Comment:
Lets move this to `addSubscriber()` and keep this method purely on the DAO logic?